src/components/screens/new-questionnaire-screen.tsx

```diff
@@ -7,52 +21,111 @@ import { KeyboardReturn } from "@mui/icons-material";
 import UserRoleUtils from "src/utils/user-role-utils";
 import type { Question, QuestionOption } from "src/types/index";
 import strings from "src/localization/strings";
+import { useLambdasApi } from "src/hooks/use-api";
+import { useSetAtom } from "jotai";
+import { errorAtom } from "src/atoms/error";
 
 /**
  * New Questionnaire Screen component
- * Includes a card (new-questionnaire-card.tsx) to create questions for a new questionnaire
  */
 const NewQuestionnaireScreen = () => {
   const adminMode = UserRoleUtils.adminMode();
-  
+  const { questionnaireApi } = useLambdasApi();
+  const [loading, setLoading] = useState(false);
+  const setError = useSetAtom(errorAtom);
   const [questionnaireTitle, setQuestionnaireTitle] = useState("");
+  const [questionnaireDescription, setQuestionnaireDescription] = useState("");
+  const [options, setOptions] = useState<Question[]>([]);
+  const [passScoreValue, setPassScoreValue] = useState(0);
 
-  const handleQuestionTitleChange = (event: ChangeEvent<HTMLInputElement>) => {
-    setQuestionnaireTitle(event.target.value);
+  /**
+   * Function to handle input change in the questionnaire title and description
+   * @param event
+   */
+  const handleQuestionnaireInputChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = event.target;
+    if (name === "title") {
+      setQuestionnaireTitle(value);
+    } else if (name === "description") {
+      setQuestionnaireDescription(value);
+    }
   };
```

Comment:
I'll expand on my previous suggestion as there is a cleaner way of doing this:

First you should have a single questionnaireState something like this:
const [questionnaire, setQuestionnaire] = useState<Questionnaire>({
    description: "",
    options: [],
    title: "",
    passScore: 0
  });
  
  Then the handler function would be like this:
   /**
   * Function to handle input change in the questionnaire title and description
   * @param event
   */
  const handleQuestionnaireInputChange = (event: ChangeEvent<HTMLInputElement>) => {
    const { name, value } = event.target;

    setQuestionnaire((prevQuestionnaire) => ({ ...prevQuestionnaire, [name]: value }));
  };
  
  Try implementing and testing this. If anything is unclear just ask

Reply:
You might need to tweak this a little bit to handle options but should still be quite simple

Reply:
I think state management is working now, but problem accured when saving to DB. It worked before. Try to fix this, but might need checking it together.